utils/dom.js

```diff
@@ -36,38 +65,40 @@ export function isHorizontalEdge( container, isReverse, collapseRanges = false )
 		return true;
 	}
 
-	// If the container is empty, the caret is always at the edge.
+	// If the container is empty, the selection is always at the edge.
 	if ( tinymce.DOM.isEmpty( container ) ) {
 		return true;
 	}
 
 	const selection = window.getSelection();
-	let range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
-	if ( collapseRanges ) {
-		range = range.cloneRange();
-		range.collapse( isReverse );
-	}
+	const range = selection.rangeCount ? selection.getRangeAt( 0 ) : null;
 
-	if ( ! range || ! range.collapsed ) {
+	if ( ! range ) {
 		return false;
 	}
 
-	const position = isReverse ? 'start' : 'end';
-	const order = isReverse ? 'first' : 'last';
-	const offset = range[ `${ position }Offset` ];
+	// Start and end offsets are always sorted small to large regardless of
+	// selection direction. If `isReverse` is does not align to selection
+	// being backward, invert the position.
+	let position = isReverse ? 'start' : 'end';
+	if ( isReverse === isSelectionForward( selection ) ) {
+		position = ( position === 'start' ) ? 'end' : 'start';
+	}
 
-	let node = range.startContainer;
+	const offset = range[ `${ position }Offset` ];
 
 	if ( isReverse && offset !== 0 ) {
 		return false;
 	}
 
+	let node = range.startContainer;
 	const maxOffset = node.nodeType === TEXT_NODE ? node.nodeValue.length : node.childNodes.length;
 
 	if ( ! isReverse && offset !== maxOffset ) {
 		return false;
 	}
 
+	const order = isReverse ? 'first' : 'last';
```

Comment:
Would this also need to invert? I haven't checked.

Reply:
>Would this also need to invert? I haven't checked.

I don't think it should need to, since this is for traversing through parents at the extent of the node, where selection direction doesn't matter.